### ISSUE TITLE
Archived geocaches incorrectly shown on map when filtering based on Cache Owner

### ISSUE THREAD

**Issue by thoger** [cgeo/cgeo#13619]
### Describe your problem!

When map filter is configured to use "Cache Owner" condition to only show caches from a specific owner, archived caches are shown on the map even when map is configured to not show archived caches.  Additionally, those caches are not shown as archived, but rather shown as active.  Only after opening of such geocache (by clicking on the geocache icon on the map and selecting "More details", or clicking the "refresh" / "reload" button), which causes the geocache to be cached on the device, c:geo correctly determines the "archives" status and removes such cache from the map.  However, this does not solve the problem permanently, and it re-appears once cached information about the geocache is removed.  A possible workaround is to store such geocache in the device, so that its status is never forgotten.

### How to reproduce?

1. Switch to map and open filter settings.  Switch to the "Advanced" filter mode.
2. In the "Status" section of the filter configuration, ensure that "Archived" is not selected.
3. Use "Add filter criteria" to add "Cache Owner" condition.  Use "Contains" operator and enter cache owner name as "Search Text".  Make sure to specify owner with an archived cache.
4. Confirm filter configuration and review geocaches shown on map.

### Actual result after these steps?

Archived caches are shown on map.  They are shown using the standard icon for the particular cache type, not greyed.  Only after opening them they disappear from the map.

### Expected result after these steps?

Archived caches are not shown.

### Reproducible

Yes

### c:geo Version

2022.10.17

### Additional Information

This may be related to the behaviour I'm seeing on the geocaching.com site map.  When looking at the map on the site with the default filter options, I do not see any archived caches.  However, after adding value for the "Hidden by" filter condition, archived caches are shown.  I wonder if cache "archived" status is included in the search results c:geo gets from the GC site and simply ignores it, assuming archived caches are never returned, or if that information is not part of the search results and hence loading of all geocache data is the only way to determine its archived status.

**Comment by thoger** [comment]
To give a more specific example, navigate on the map to the area near the Geocaching Headquarters and look for Geocaching HQ caches.  See this geocaching.com map link:

https://www.geocaching.com/play/map?lat=47.6505&lng=-122.3553&zoom=16&asc=true&sort=distance&st=N+47%C2%B0+39.337%27+W+122%C2%B0+21.632%27&ot=coords&hb=Geocaching+HQ

On the map, you should see archived caches [GC2AD97](https://coord.info/GC2AD97) and [GC4GWB5](https://coord.info/GC4GWB5), and also couple of archived event caches.

Open the same area in c:geo, set Cacher Owner condition to "Geocaching HQ" (without quotes) in filter settings, and see those caches shown using the standard green icon for active traditional caches.

**Comment by Lineflyer** [comment]
Bug confirmed and reproducible on current `master`. Thanks for your excellent description.

It seems as if we either don`t handle the situation, that archived caches are part of the result or there is a bug with that.

Its even easier to reproduce by just searching for caches owned by "Geocaching HQ" using "search with filter":
[GC2AD97](https://coord.info/GC2AD97) and [GC4GWB5](https://coord.info/GC4GWB5) are returned as active until you load the details.

**Comment by ztNFny** [comment]
Can also be reproduced by opening any cache, tapping the owners name, selecting "hidden by"

**Comment by fm-sys** [comment]
The following data is returned by the API, the field `cacheStatus` sounds interesting... The fix should be a simple one-liner. Will target to `release`

![grafik](https://user-images.githubusercontent.com/64581222/205314184-6794e056-c6ef-46a8-a595-695c408970a8.png)
